### PR TITLE
[AzureMonitorExporter] Collect AKS attach rate numerator

### PR DESCRIPTION
**Pull request by rajkumar-rangaraj**
Based on https://github.com/Azure/azure-sdk-for-java/pull/37123

### CODE REVIEW

**Review by cijothomas — APPROVED** [review]

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/EnvironmentVariableConstants.cs [hunk 1]
```diff
@@ -56,6 +56,16 @@ internal static class EnvironmentVariableConstants
         /// </summary>
         public const string WEBSITE_SITE_NAME = "WEBSITE_SITE_NAME";
 
+        /// <summary>
+        /// INTERNAL ONLY. Used by Statsbeat to get the Kubernetes Service Host.
+        /// </summary>
+        public const string KUBERNETES_SERVICE_HOST = "KUBERNETES_SERVICE_HOST";
```
Comment (heyams) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/EnvironmentVariableConstants.cs:62]: this is available for any kubernetes, including AKS. 
it's incorrect to send rpId = aks when it's kubernetes. it's not possible to collect denominator for kubernetes at all.
i think for now we would send if AKS_ARM_NAMESPACE_ID when it has value (auto-instrumentation). 
PM will figure out the manual instrumentation numerator by working with AKS directly. We can follow up with them to find out.

Comment (rajkumar-rangaraj) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/EnvironmentVariableConstants.cs:62]: Are you asking to remove this string _resourceProvider = "aks";?

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/Statsbeat/AzureMonitorStatsbeat.cs [hunk 2]
```diff
@@ -189,6 +189,15 @@ private void SetResourceProviderDetails(IPlatform platform)
...
             }
 
+            var kubernetesSrvHost = platform.GetEnvironmentVariable(EnvironmentVariableConstants.KUBERNETES_SERVICE_HOST);
+            if (kubernetesSrvHost != null)
+            {
+                _resourceProvider = "aks";
+                _resourceProviderId = platform.GetEnvironmentVariable(EnvironmentVariableConstants.AKS_ARM_NAMESPACE_ID) ?? "unknown";
+
+                return;
+            }
```
Comment (heyams) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/Statsbeat/AzureMonitorStatsbeat.cs:199]: var aksArmNamespaceId = platform.GetEnvironmentVariable(EnvironmentVariableConstants.AKS_ARM_NAMESPACE_ID);
if (aksArmNamespaceId != null)
{
    _resourceProvider = "aks";
    _resourceProviderId = aksArmNamespaceId

    return;
}

Comment (heyams) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/Statsbeat/AzureMonitorStatsbeat.cs:199]: @rajkumar-rangaraj i'm proposing this for now.

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/EnvironmentVariableConstants.cs (hunk 1 above)
Comment (heyams) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/EnvironmentVariableConstants.cs:62]: https://github.com/Azure/azure-sdk-for-net/pull/39663#discussion_r1379122997

**Pull request merged**